src/api/app/controllers/webui/project_controller.rb

```diff
@@ -526,8 +526,8 @@ def load_project_info
 
     reqs = @project.open_requests
     @requests = (reqs[:reviews] + reqs[:targets] + reqs[:incidents] + reqs[:maintenance_release]).sort!.uniq
-    @incoming_requests = RequestsFinder.new.incoming(reqs.values.sum, @project.name)
-    @outgoing_requests = RequestsFinder.new.outgoing(reqs.values.sum, @project.name)
+    @incoming_requests_size = RequestsFinder.new.incoming(reqs.values.sum, @project.name).size
+    @outgoing_requests_size = RequestsFinder.new.outgoing(reqs.values.sum, @project.name).size
```

Comment:
Now you load a lot of requests into memory into an instance variable array to count them. But your database supports counting with [`.count`](https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-count). 

https://medium.com/ruby-daily/comparing-count-length-and-size-in-ruby-on-rails-e061840bc19e

Reply:
I will narrow down the scope of the query to count the db records only and return the number at that level. Yeah it will be much more efficient. The only counter side is that we leave out the option to reuse the query for wider uses (for instance to fetch and render only incoming requests), but I guess we can still write a specific query for that afterwards if and when it will be needed.